rust/agama-network/src/model.rs

```diff
@@ -0,0 +1,439 @@
+//! Representation of the network configuration
+//!
+//! * This module contains the types that represent the network concepts. They are supposed to be
+//! agnostic from the real network service (e.g., NetworkManager).
+use uuid::Uuid;
+
+use crate::error::NetworkStateError;
+use std::{fmt, net::Ipv4Addr};
+
+#[derive(Default)]
+pub struct NetworkState {
+    pub devices: Vec<Device>,
+    pub connections: Vec<Connection>,
+}
+
+impl NetworkState {
+    /// Returns a NetworkState struct with the given devices and connections.
+    ///
+    /// * `devices`: devices to include in the state.
+    /// * `connections`: connections to include in the state.
+    pub fn new(devices: Vec<Device>, connections: Vec<Connection>) -> Self {
+        Self {
+            devices,
+            connections,
+        }
+    }
+
+    /// Get device by name
+    ///
+    /// * `name`: device name
+    pub fn get_device(&self, name: &str) -> Option<&Device> {
+        self.devices.iter().find(|d| d.name == name)
+    }
+
+    /// Get connection by UUID
+    ///
+    /// * `uuid`: connection UUID
+    pub fn get_connection(&self, uuid: Uuid) -> Option<&Connection> {
+        self.connections.iter().find(|c| c.uuid() == uuid)
+    }
+
+    /// Get connection by UUID as mutable
```

Comment:
this is kind of inconsistent as get_device_mut missing

Reply:
It is missing because it is not needed. We do not allow changing anything for the devices yet. Once we implement support for that, we will add that method.